src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -402,6 +402,20 @@ private BoundExpression VisitBinaryOperator(BinaryOperatorKind opKind, MethodSym
             }
         }
 
+        private BoundExpression PromoteEnumOperand(BoundExpression operand, BoundExpression loweredOperand, TypeSymbol promotedType, bool isChecked)
+        {
+            var literal = operand as BoundLiteral;
+            if (literal != null)
+            {
+                // for compat reasons enum literals are directly promoted into underlying values
+                return Visit(literal.Update(literal.ConstantValue, promotedType));
```

Comment:
No, the loweredOperand has been visited and now contains conversions to enum and back to the underlying type. We are visiting the literal with tweaked type directly here to skip the conversions.